### PR TITLE
guard against invalid sprite access crashes

### PR DESCRIPTION
**Pull request by Chaosmeister**
ride->num_vehicles not matching ride->vehicles[] in ride_prepare_breakdown() and window_ride_maintenance_dropdown():

Fixes: #3346

### CODE REVIEW

File: src/openrct2/ride/ride.c [hunk 1]
```diff
@@ -2412,6 +2412,18 @@ static sint32 ride_get_new_breakdown_problem(rct_ride *ride)
     return BREAKDOWN_BRAKES_FAILURE;
 }
 
+void choose_random_train_safe(rct_ride * ride)
```
Comment (janisozaur) [inline src/openrct2/ride/ride.c]: `static`

**Comment by janisozaur** [comment]
Please also fix your commit message: https://github.com/OpenRCT2/OpenRCT2/wiki/Commit-Messages

File: src/openrct2/ride/ride.c (hunk 1 above)
Comment (IntelOrca) [inline src/openrct2/ride/ride.c]: This isn't the best name, it doesn't mention breakdown.

**Comment by janisozaur** [comment]
And also update network version

File: src/openrct2/windows/ride.c [hunk 2]
```diff
@@ -3794,14 +3794,16 @@ static void window_ride_maintenance_dropdown(rct_window *w, rct_widgetindex widg
...
-                    } while ((spriteId = vehicle->next_vehicle_on_train) != 0xFFFF);
+                    if (spriteId != SPRITE_INDEX_NULL) {
+                        do {
+                            vehicle = GET_VEHICLE(spriteId);
+                            vehicle->update_flags &= ~(
+                                VEHICLE_UPDATE_FLAG_BROKEN_CAR |
+                                VEHICLE_UPDATE_FLAG_7 |
+                                VEHICLE_UPDATE_FLAG_BROKEN_TRAIN
+                                );
+                        } while ((spriteId = vehicle->next_vehicle_on_train) != 0xFFFF);
```
Comment (janisozaur) [inline src/openrct2/windows/ride.c]: This already checks for `SPRITE_INDEX_NULL`, is there no better way to format this loop?

File: src/openrct2/ride/ride.c [hunk 3]
```diff
@@ -2412,6 +2412,18 @@ static sint32 ride_get_new_breakdown_problem(rct_ride *ride)
     return BREAKDOWN_BRAKES_FAILURE;
 }
 
+static void choose_random_train_to_breakdown_safe(rct_ride * ride)
+{
+    ride->broken_vehicle = scenario_rand() % ride->num_vehicles;
+
+    // Prevent crash caused by acessing SPRITE_INDEX_NULL on hacked rides.
```
Comment (janisozaur) [inline src/openrct2/ride/ride.c]: typo, should be: `accessing`

File: src/openrct2/windows/ride.c (hunk 2 above)
Comment (Chaosmeister) [inline src/openrct2/windows/ride.c]: Good eyes. I somehow missed that.

**Pull request merged**